tests/ui/test_api.py

```diff
@@ -64,3 +69,29 @@ def test_dbcommit(app):
     })
 
     record.dbcommit()
+    assert DocumentRecord.get_record_by_pid(
+        '1')['title'] == 'The title of the record'
+
+
+def test_reindex(app, db, client):
+    """Test record reindex."""
+    record = DocumentRecord.create({
+        "pid": "100",
+        "title": "The title of the record"
+    })
+    db.session.commit()
+
+    record.reindex()
+
+    # Wait for indexing
+    time.sleep(1)
```

Comment:
you can force to flush the index instead. Please check the rero-ils Patron indexer for example. Sleep is not safe.